Updates responsive embeds with new class names and CSS variables

mdo opened the pull request:
- Renames .embed-responsive-XbyY classes to .embed-responsive-XxY
- Simplifies the Sass map to just be key and value, ratio and percentage
- Builds .embed-responsive-* modifiers with CSS variables
- Updates docs to show power of CSS variables
- Add notes to the Migration guide

Fixes #31531.

---

<img width="829" alt="Screen Shot 2020-09-21 at 2 15 25 PM" src="https://user-images.githubusercontent.com/98681/93822369-fcdb5700-fc14-11ea-8bcd-7ca723d21d93.png">

<img width="827" alt="Screen Shot 2020-09-21 at 2 15 30 PM" src="https://user-images.githubusercontent.com/98681/93822372-fe0c8400-fc14-11ea-8d21-b56a4b83ae28.png">

mdo committed:
Updates responsive embeds with new class names and CSS variables

- Renames .embed-responsive-XbyY classes to .embed-responsive-XxY
- Simplifies the Sass map to just be key and value, ratio and percentage
- Builds .embed-responsive-* modifiers with CSS variables
- Updates docs to show power of CSS variables
- Add notes to the Migration guide

mdo commented on scss/_variables.scss:
```diff
@@ -378,22 +378,10 @@ $transition-collapse:         height .35s ease !default;
+  "1x1": 100%,
+  "4x3": 75%,
+  "16x9": 56.25%,
+  "21x9": 42.857142%
```
`calc(21/9%)` doesn't work. We'd have to do something like `calc(calc(9/21) * 100%)`, which doesn't feel more readable to me.

MartijnCuppens replied:
`calc(9 / 21 * 100%)` or `calc(900% / 21)` should do the trick.

ffoodd replied:
Agreed, I prefer self-explanatory values even if they can be simpler another way. The percentage result looks like a magical number but isn't.

MartijnCuppens commented on site/content/docs/5.0/helpers/embed.md:
```diff
@@ -26,28 +28,35 @@ Wrap any embed like an `<iframe>` in a parent element with `.embed-responsive` a
-<!-- 4:3 aspect ratio -->
-<div class="embed-responsive embed-responsive-4by3">
-  <iframe class="embed-responsive-item" src="..."></iframe>
+<div class="embed-responsive embed-responsive-16x9" style="width: 10rem;">
```
I would move the `style` to the docs css, this might get copy-pasted.

MartijnCuppens commented:
Thoughts about renaming `embed-responsive` to `ratio`?

mdo commented:
Think that'd be fine @MartijnCuppens. We'd change the `.responsive-embed-item` too, and keep the nested selectors, right?

mdo committed:
Rename the classes and docs page, update everything accordingly

mdo committed:
Move styles to docs CSS

mdo committed:
Add example of responsive behavior for aspect ratios

mdo commented:
Done, and done, and done! Latest commits rename it to `.ratio` and move inline styles to the docs CSS. Also added an example of using breakpoints to change the CSS variable for responsive ratios.

ffoodd reviewed (approved):
LGTM!

MartijnCuppens commented:
![Very nice](https://media.giphy.com/media/10uEX5kfeodYgo/giphy.gif)

One last thought:

We could simplify this to just `> *`:
https://github.com/twbs/bootstrap/blob/f407d32134c8e6b09644c12fa51af56e5a135ce8/scss/helpers/_ratio.scss#L13-L17
It feels a bit inconsistent that we would require `.ratio-item` on a `<div>` or an `<img />`, but not on an `<iframe>` or a `<video>`

mdo committed:
Updates to .ratio selectors and docs

- Remove group selector for ratio items
- Drop the .ratio-item entirely
- Update docs to explain updated approach
- Update Migration guide to reflect the latest

mdo commented:
Yup, good idea—just pushed some changes.Done!

MartijnCuppens reviewed (approved):
:shipit:
